security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java

```diff
@@ -0,0 +1,368 @@
+					logger.log(Level.INFO, "Found Account ID: {0}", currentAccountId);
+
+					if (entityType.equals("Application Instance") && currentAccountId.equals(itemId)) {
+						logger.log(Level.FINE, "Matched account ID...");
+
+						currentAccount.setValidToDate(entityEndDate);
+						// must clear account data in order for modify to work
+						// (due to partial child data)
+						currentAccount.setAccountData(null);
+						provisioningService.modify(currentAccount);
+						logger.log(Level.FINE, "Updated user account.");
+
+						logger.log(Level.FINE, "Removing token from the database...");
+
+						dbConnection = getDBConnection();
+						PreparedStatement deletePS = dbConnection.prepareStatement(EXTENSION_DELETE_QUERY);
+						deletePS.setString(1, eaUUID);
+						deletePS.execute();
+
+						deletePS.close();
+						dbConnection.close();
+
+						logger.log(Level.INFO, "Account extension complete.");
+
+						restOutput = "Account access has been extended. You can now close this browser window.";
+					} else {
+						List<EntitlementInstance> applicationEnitlementGrants = currentAccount.getEntitlementGrants();
+
+						for (EntitlementInstance currentEntitlementInstance : applicationEnitlementGrants) {
+							String currentEntitlementInstanceKey = String
+									.valueOf(currentEntitlementInstance.getEntitlementInstanceKey());
+							logger.log(Level.INFO, "Found Entitlement Instance Key: {0}",
+									currentEntitlementInstanceKey);
+
+							if (currentEntitlementInstanceKey.equals(itemId)) {
+								logger.log(Level.FINE, "Matched entitlement instance key...");
+
+								currentEntitlementInstance.setValidToDate(entityEndDate);
+								provisioningService.updateEntitlement(currentEntitlementInstance);
+								logger.log(Level.FINE, "Updated user entitlement instance.");
+
+								logger.log(Level.FINE, "Removing token from the database...");
+								dbConnection = getDBConnection();
+								PreparedStatement deletePS = dbConnection.prepareStatement(EXTENSION_DELETE_QUERY);
+								deletePS.setString(1, eaUUID);
+								deletePS.execute();
+
+								deletePS.close();
+								dbConnection.close();
+
+								logger.log(Level.INFO, "Entitlement extension complete.");
+
+								restOutput = "Entitlement access has been extended. You can now close this browser window.";
+							}
+						}
+					}
+				}
+			}
+
+			if (resultValuesList.isEmpty()) {
+				logger.log(Level.WARNING, "The extension token is no longer present in the DB.");
+				restOutput = "Error: Invalid or expired extension token.";
+			}
+
+		} catch (Exception e) {
+			logger.log(Level.SEVERE, "Severe exception encountered: ", e);
+		} finally {
+			try {
+				if (resultSet != null && !resultSet.isClosed())
+					resultSet.close();
+			} catch (SQLException e) {
+				logger.log(Level.WARNING, "Error on RS termination: ", e);
+			}
+			try {
+				if (preparedStatement != null && !preparedStatement.isClosed())
```

Comment:
Use curly braces for single line blocks.

Reply:
Fixed.